src/visual.ts

```diff
@@ -16,8 +16,9 @@ import DataView = powerbi.DataView;
 import * as vega from 'vega';
 import * as vl from 'vega-lite';
 import { VisualSettings } from "./settings";
+import { Dict } from "vega-lite/build/src/util";
 
-interface BarChartDataPoint {
+interface BarChartDataPoint extends Dict<unknown> {
```

Comment:
@domoritz is this how this is intended to be used?

Reply:
Otherwise [this line](https://github.com/microsoft/vegalite-for-powerbi/blob/3be526224828a560b8a0cd36314003e49f347f68/src/visual.ts#L98) causes a compilation error

Reply:
Hmm, ideally that would not be necessary. We can roll back the Vega-Lite change but it was what eslint recommends. See https://github.com/typescript-eslint/typescript-eslint/blob/v3.0.2/packages/eslint-plugin/docs/rules/ban-types.md. They recommend using `Record<string, unknown>`, which is the same as `Dict<unknown>`. What do you think make sense?

Reply:
@domoritz It seems that an interface like this one below should be compatible with `Dict<unknown>` :
```
interface BarChartDataPoint {
    value: PrimitiveValue;
    category: string;
}
```
But it is not because `Index signature is missing in type 'BarChartDataPoint[]'`. However, it does work with a cast to `Dict<any>`. Perhaps it's one place where `object` makes sense, and you can make an eslint exception. Let's see if other users get the latest vega-lite and encounter compiler issues.

Reply:
Hmm, maybe object maybe object makes sense then. Or what type has object properties but does not require an index signature?

Reply:
A literal type has an index signature, so if we go with `Dict<unknown>` then this works:
```
type BarChartDataPoint = {
    value: PrimitiveValue;
    category: string;
}
```
According to [this comment](https://github.com/microsoft/TypeScript/issues/32263#issuecomment-508750932):
> object literal types have implicit index signatures, while interfaces do not

According to [the handbook](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-aliases):
> you should always use an interface over a type alias if possible.

I'm not sure of any type other than object.

Reply:
Let's revert back to `object` then. Can you send a pull request?